Classes/Private/Manager/Audio/OCTAudioEngine.m

```diff
@@ -127,6 +140,102 @@ - (BOOL)isAudioRunning:(NSError **)error
     return running;
 }
 
+#pragma mark - Buffer Management
+- (void)provideAudioFrames:(const int16_t *)pcm sample_count:(size_t)sample_count channels:(uint8_t)channels sample_rate:(uint32_t)sample_rate
+{
+    // To Do: copy audio frames into buffer.
+}
+
+#pragma mark - Call Backs
+
+- (BOOL)registerInputCallBack:(NSError **)error;
+{
+    AURenderCallbackStruct callbackStruct;
+    callbackStruct.inputProc = inputRenderCallBack;
+    callbackStruct.inputProcRefCon = _ioUnit;
```

Comment:
inputProcRefCon is used is for passing the reference of any object you would like to manage/do things with during a callback. I originally intended to pass only the audio unit since I figure we didn't need the pointer of the object itself. 

However looking at it again, I need to call `[OCTToxAV sendAudioFrames]` later so I will need to pass `self` too for input as well. I'll change this.